packages/db/src/connection.ts

```diff
@@ -0,0 +1,129 @@
+import Knex from 'knex';
+import { getSecretConnectionConfig, getEnvConnectionConfig } from './config';
+
+import { envConectionConfigObject, knexSecretConnectionConfigObject } from './types';
+
+export interface knexConnectionConfigObject {
+  host: string,
+  username: string,
+  password: string,
+  database?: string,
+}
+
+interface knexConfigObject {
+  client: string,
+  connection: knexConnectionConfigObject,
+  acquireConnectionTimeout: number,
+  asyncStackTraces: boolean,
+  debug: boolean,
+  migrations?: {
+    directory: string
+  }
+}
+
+interface knexEnvironmentObject extends NodeJS.ProcessEnv {
+  KNEX_DEBUG?: string
+  KNEX_ASYNC_STACK_TRACES?: string,
+}
+
+/**
+* @summary Given a knexEnvironmentObject and a Knex.PgConnectionConfig, returns
+* a Knex instance with those confiugrations applied
+*
+* @param {Knex.PgConnectionConfig} connectionConfig - Knex connection configuration
+* @param {knexEnvironmentObject} env - environment object with knex configuration keys
+* @returns {Knex} - configured Knex instance
+*/
+const getConfiguredKnex = (
+  connectionConfig: Knex.PgConnectionConfig,
+  env: knexEnvironmentObject
+): Knex => {
+  let knexConfig = {
```

Comment:
If you update this line to: `const knexConfig: Knex.Config = {` (and remove the `as knexConfigObject` on line 47, it reveals that there's an issue with the `acquireConnectionTimeout` assignment. It also prevents us from having to define our own config type.

Reply:
72d40c3d1